synapse/rest/client/v2_alpha/account.py

```diff
@@ -107,21 +116,35 @@ async def on_POST(self, request):
 
             raise SynapseError(400, "Email not found", Codes.THREEPID_NOT_FOUND)
 
+        # Get the email addresses of the user from database
+        # The user can own more than one address. Lookup for the right address.
+        # The email will be sent to the stored address.
+        # It should prevent potential account hijack via password reset form,
+        # if some compare algorithm are not exactly.
+        addresses = await self.account_validity_handler._get_email_addresses_for_user(
+            existing_user_id
+        )
+        for address in addresses:
+            if address == email:
+                email_from_database = address
+        if not email_from_database:
+            raise SynapseError(400, "Email not found", Codes.THREEPID_NOT_FOUND)
+
```

Comment:
I'm afraid I've forgotten what we said in previous iterations of this PR, but looking at this again: 

surely this is redundant and we can just send the password reset to the canonicalised address `email`? `get_user_id_by_threepid` will fail unless it exactly matches the email in the database?

Reply:
Yes, it's a double check. I'm extra careful and a little paranoid.
Should I remove the double check?

Reply:
I think it adds complication and confusion without adding much value. Yes, please remove it.